gson/gson-bundle/pom.xml

```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Licensed to the Apache Software Foundation (ASF) under one or more
+    contributor license agreements.  See the NOTICE file distributed with
+    this work for additional information regarding copyright ownership.
+    The ASF licenses this file to You under the Apache License, Version 2.0
+    (the "License"); you may not use this file except in compliance with
+    the License.  You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <artifactId>jclouds-project</artifactId>
+    <groupId>org.apache.jclouds</groupId>
+    <version>2.2.0-SNAPSHOT</version>
+    <relativePath>../../project/pom.xml</relativePath>
+  </parent>
+
+  <artifactId>jclouds-gson</artifactId>
+  <name>jclouds-gson OSGi bundle</name>
+  <description>jclouds-gson OSGi bundle with gson internal packages relocated</description>
+  <packaging>bundle</packaging>
+
+  <properties>
+    <jclouds.osgi.export>
+      org.jclouds.json.gson.internal*;version=${project.version};-noimport:=true,
+      !com.google.gson.internal*,
+      com.google.gson*;version=${project.version};-noimport:=true,
+    </jclouds.osgi.export>
+    <jclouds.osgi.import>*</jclouds.osgi.import>
+  </properties>
+
+  <dependencies>
```

Comment:
Argh - this is what you meant by having to split the shading and the bundling up. Grr. I guess it sort-of makes sense since the shading has to happen before the bundle manifest generation.

Out of curiosity, did you try binding the shade plugin to a phase _before_ `package`, e.g. [`prepare-package`](https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html#Lifecycle_Reference)? Definitely not important if this ends up working, I should add!

Reply:
Yep, I tried that and did not work, also binding the bundle plugin after shade explicitly, same thing.

Reply:
_sigh_. Thanks for explaining!